iroha/src/merkle.rs

```diff
@@ -15,14 +15,12 @@ impl MerkleTree {
         }
     }
 
-    /// Builds a Merkle Tree from sorted array of `ValidBlocks`.
-    //TODO: should we check or sort blocks here?
-    pub fn build(&mut self, blocks: &[&ValidBlock]) {
-        //hm, can we write map(ValidBlock::hash) in Rust?
-        let mut nodes: std::collections::VecDeque<Node> = blocks
-            .iter()
-            .map(|block| Node::Leaf { hash: block.hash() })
-            .collect();
+    /// Builds a Merkle Tree from an array of `Hash` values of `Block` or `Transaction`.
```

Comment:
Maybe abstract over `Block` and `Transaction` even more:
`values. For example of Blocks and Transactions hashes.`?